services/directions.js

```diff
@@ -0,0 +1,151 @@
+'use strict';
+
+var v = require('./service-helpers/validator');
+var createServiceFactory = require('./service-helpers/create-service-factory');
+
+/**
+ * Directions API service.
+ */
+var Directions = {};
+
+/**
+ * Get directions.
+ *
+ * See the [Mapbox Directions API](https://www.mapbox.com/api-documentation/#directions).
+ *
+ * @param {Object} config
+ * @param {'driving-traffic'|'driving'|'walking'|'cycling'} [config.profile=driving]
+ * @param {Array.<Object>} config.way_points - An ordered array of objects with `latitude`
```

Comment:
Flow-type annotations should work, so `Array<Object>` (no need for the period). In fact, we may want to see what happens in the doc if we flesh out this annotation, e.g. `Array<{ longitude: number, latitude: number, approach?: 'unrestried' | 'curb', ...}>`

Reply:
The `{key?: value}`  doesn't seem to work with it, I thinking of using `typedef` to document `wayPoints`.